src/shareddata.cpp

```diff
@@ -182,10 +184,17 @@ SharedData::initialize(const char *tmpDir = NULL,
   } else {
     struct stat statbuf;
     while (1) {
+      bool initialized = false;
       Util::lockFile(PROTECTED_SHM_FD);
       JASSERT(fstat(PROTECTED_SHM_FD, &statbuf) != -1) (JASSERT_ERRNO);
+      initialized = sharedDataHeader->initialized;
       Util::unlockFile(PROTECTED_SHM_FD);
-      if (statbuf.st_size > 0) {
+      // If we got here, it implies that some other peer won the race
```

Comment:
Looks great, but please add a FIXME or TODO for the scenario when the two processes are synchronized and hit the check at the same time, where your recommendation of SysV semaphores/condition variables would be a more robust check.

Thanks,
~p